crates/core/src/sql/compiler.rs

```diff
@@ -332,6 +332,23 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn compile_not_eq() -> ResultTest<()> {
+        let (db, _tmp) = make_test_db()?;
+
+        // Create table [test] with cols [a, b] and index on [b].
+        db.create_table_for_test(
+            "test",
+            &[("a", AlgebraicType::U64), ("b", AlgebraicType::U64)],
+            &[(1.into(), "b")],
+        )?;
+
+        let tx = db.begin_tx();
+        // Should work with any qualified field.
+        assert!(compile_sql(&db, &tx, "select * from test where test.b <> 3").is_ok());
```

Comment:
Could you modify this test to be something like
```
select * from test where a = 1 and b <> 3
```
where there is also an index on `a`, so that we make sure we still select the index scan on `a`.